### PR TITLE
[libclangex] Don't dlopen

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Review by giordano — APPROVED** [review]
Looks good, but please use `[skip build]` when squash-merging :slightly_smiling_face:

**Comment by vchuravy** [comment]
Wait we do need to rebuild the JLL? Does that work with [skip build]?

**Comment by giordano** [comment]
Yes, the JLL is always regenerated.  `[skip build]` is useful in cases like this one were we want to update the JLL wrapper, not the artifacts.

File: L/libclangex/build_tarballs.jl [hunk 1]
```diff
@@ -40,7 +40,10 @@ function configure(julia_version, llvm_version)
 
     # The products that we will ensure are always built
     products = Product[
-        LibraryProduct("libclangex", :libclangex)
+        # Life is horrible and we can't add Clang_jll as a dependency, thus
+        # loading the library will always fail. We can fix this in the consumer
+        # so don't dlopen
```
Comment (Gnimuc) [inline L/libclangex/build_tarballs.jl:45]: magic!

**Pull request merged**